src/assets/Generator.Shared/Utf8JsonWriterExtensions.cs

```diff
@@ -140,5 +140,26 @@ public static void WriteObjectValue(this Utf8JsonWriter writer, object value)
                     throw new NotSupportedException("Not supported type " + value.GetType());
             }
         }
+
+        public static void WriteBinaryDataValue(this Utf8JsonWriter writer, BinaryData value)
+        {
+            string inputValue = value.ToString();
+            // First we check if it is possible to serialize the value as json value.
+            if (inputValue.StartsWith("{", StringComparison.InvariantCultureIgnoreCase) && (inputValue.EndsWith("}", StringComparison.InvariantCultureIgnoreCase) || inputValue.EndsWith($"}}{Environment.NewLine}", StringComparison.InvariantCultureIgnoreCase)) ||
+                inputValue.StartsWith("[", StringComparison.InvariantCultureIgnoreCase) && (inputValue.EndsWith("]", StringComparison.InvariantCultureIgnoreCase) || inputValue.EndsWith($"]{Environment.NewLine}", StringComparison.InvariantCultureIgnoreCase)) ||
```

Comment:
consider to have a perf test

Reply:
I tested the performance using implementation with try-catch and without try-catch respectively and below is the result.

- When the BinaryData is a json object

|         Method |           Job |       Runtime |       Mean |    Error |   StdDev |     Median |        Min |        Max |  Gen 0 | Allocated |
|--------------- |-------------- |-------------- |-----------:|---------:|---------:|-----------:|-----------:|-----------:|-------:|----------:|
| TestWithoutTry |      .NET 6.0 |      .NET 6.0 |   920.7 ns |  2.85 ns |  2.53 ns |   920.4 ns |   917.1 ns |   924.7 ns | 0.1574 |      1 KB |
|    TestWithTry |      .NET 6.0 |      .NET 6.0 |   952.4 ns | 10.76 ns | 10.07 ns |   954.7 ns |   937.2 ns |   968.7 ns | 0.1574 |      1 KB |
| TestWithoutTry | .NET Core 3.1 | .NET Core 3.1 | 1,576.0 ns | 23.19 ns | 21.69 ns | 1,566.8 ns | 1,551.4 ns | 1,621.7 ns | 0.1583 |      1 KB |
|    TestWithTry | .NET Core 3.1 | .NET Core 3.1 | 1,618.4 ns | 33.16 ns | 94.60 ns | 1,568.3 ns | 1,541.2 ns | 1,862.0 ns | 0.1583 |      1 KB |

- When the BinaryData is a string and we correctly treat it as string

|         Method |           Job |       Runtime |     Mean |   Error |  StdDev |      Min |      Max |  Gen 0 |  Gen 1 | Allocated |
|--------------- |-------------- |-------------- |---------:|--------:|--------:|---------:|---------:|-------:|-------:|----------:|
| TestWithoutTry |      .NET 6.0 |      .NET 6.0 | 201.0 ns | 0.62 ns | 0.55 ns | 200.1 ns | 202.1 ns | 0.1013 | 0.0002 |     848 B |
|    TestWithTry |      .NET 6.0 |      .NET 6.0 | 201.5 ns | 0.75 ns | 0.70 ns | 200.6 ns | 203.2 ns | 0.1013 | 0.0002 |     848 B |
| TestWithoutTry | .NET Core 3.1 | .NET Core 3.1 | 312.0 ns | 1.46 ns | 1.30 ns | 309.8 ns | 314.2 ns | 0.1020 |      - |     856 B |
|    TestWithTry | .NET Core 3.1 | .NET Core 3.1 | 333.8 ns | 3.62 ns | 3.21 ns | 328.7 ns | 338.1 ns | 0.1020 |      - |     856 B |

Based on above results, we can find that if we add a try-catch in this method, there will a little performance problem(however, sometimes the time are even very close to each other), but it can ensure that we could correctly handle the situation that a binary data is actually a string but we see it as a json. To be honest, I think this kind of case will be very rare, but we don't have the confidence to say it will never exist, therefore I think it's worth sacrificing a little performance here.

Reply:
How big was the underlying byte array?

Reply:
1. Vaule: `{"foo":"bar","numbers":[1,2,3,4]}`, Length: 33
2. Vaule: `abcdef`, Length: 6